src/terminal_chat_console.h

```diff
@@ -0,0 +1,114 @@
+/*
+Minetest
+Copyright (C) 2015 est31 <MTest31@outlook.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#ifndef TERMINAL_CHAT_CONSOLE_H
+#define TERMINAL_CHAT_CONSOLE_H
+
+#include "chat.h"
+#include "threading/thread.h"
+#include "chat_interface.h"
+#include "log.h"
+
+#include <sstream>
+
+class TermLogOutput : public ILogOutput {
+public:
+
+	void logRaw(LogLevel lev, const std::string &line)
+	{
+		queue.push_back(std::make_pair(lev, line));
+	}
+
+	virtual void log(LogLevel lev, const std::string &combined,
+		const std::string &time, const std::string &thread_name,
+		const std::string &payload_text)
+	{
+		std::ostringstream os(std::ios_base::binary);
+		os << time << ": [" << thread_name << "]" << payload_text;
```

Comment:
I think there should be a space character after the closing bracket otherwise you end up with the message appended to `[ << thread_name << ]` like this : https://lut.im/HeBDQBMbhW/OVPvnrqbwLYQz3xm.png